cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -55,18 +56,25 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	mf := otelmetrics.NewFactory(s.telset.MeterProvider)
 	baseFactory := mf.Namespace(metrics.NSOptions{Name: "jaeger"})
 	queryMetricsFactory := baseFactory.Namespace(metrics.NSOptions{Name: "query"})
-	f, err := jaegerstorage.GetStorageFactory(s.config.Storage.TracesPrimary, host)
+
+	f, err := jaegerstorage.GetStorageFactoryV2(s.config.Storage.TracesPrimary, host)
 	if err != nil {
 		return fmt.Errorf("cannot find primary storage %s: %w", s.config.Storage.TracesPrimary, err)
 	}
 
-	spanReader, err := f.CreateSpanReader()
+	traceReader, err := f.CreateTraceReader()
+	if err != nil {
+		return fmt.Errorf("cannot create trace reader: %w", err)
+	}
+
+	spanReader, err := factoryadapter.GetV1Reader(traceReader)
 	if err != nil {
 		return fmt.Errorf("cannot create span reader: %w", err)
 	}
 
 	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, queryMetricsFactory)
 
+	// TODO: CreateDependencyReader currently not implemented
```

Comment:
@yurishkuro `CreateDependencyReader` isn't currently path of the v2 storage factory interface - how should we proceed here?